controller/src/main/java/io/pravega/controller/task/Stream/StreamTransactionMetadataTasks.java

```diff
@@ -320,51 +324,66 @@ public Void initializeStreamWriters(final String commitStreamName, final EventSt
+                            });
+
+                    // Step 3. Create txn node in the store.
+                    CompletableFuture<VersionedTransactionData> txnFuture = addIndex.thenComposeAsync(ignore ->
+                            streamMetadataStore.createTransaction(scope, stream, txnId, lease, maxExecutionPeriod,
+                                    scaleGracePeriod, ctx, executor), executor).whenComplete((v, e) -> {
+                        if (e != null) {
+                            log.debug("Txn={}, failed creating txn in store", txnId);
+                        } else {
+                            log.debug("Txn={}, created in store", txnId);
+                        }
+                    });
 
-        // Step 4. Notify segment stores about new txn.
-        CompletableFuture<List<Segment>> segmentsFuture = txnFuture.thenComposeAsync(txnData ->
-                streamMetadataStore.getActiveSegments(scope, stream, txnData.getEpoch(), ctx, executor), executor);
-
-        CompletableFuture<Void> notify = segmentsFuture.thenComposeAsync(activeSegments ->
-                notifyTxnCreation(scope, stream, activeSegments, txnId), executor).whenComplete((v, e) ->
-                // Method notifyTxnCreation ensures that notification completes
-                // even in the presence of n/w or segment store failures.
-                log.debug("Txn={}, notified segments stores", txnId));
-
-        // Step 5. Start tracking txn in timeout service
-        return notify.whenCompleteAsync((result, ex) -> {
-            int version = 0;
-            long executionExpiryTime = System.currentTimeMillis() + maxExecutionPeriod;
-            if (!txnFuture.isCompletedExceptionally()) {
-                version = txnFuture.join().getVersion();
-                executionExpiryTime = txnFuture.join().getMaxExecutionExpiryTime();
-            }
-            timeoutService.addTxn(scope, stream, txnId, version, lease, executionExpiryTime, scaleGracePeriod);
-            log.debug("Txn={}, added to timeout service on host={}", txnId, hostId);
-        }, executor).thenApplyAsync(v -> new ImmutablePair<>(txnFuture.join(), segmentsFuture.join()), executor);
+                    // Step 4. Notify segment stores about new txn.
+                    CompletableFuture<List<Segment>> segmentsFuture = txnFuture.thenComposeAsync(txnData ->
+                            streamMetadataStore.getActiveSegments(scope, stream, txnData.getEpoch(), ctx, executor), executor);
+
+                    CompletableFuture<Void> notify = segmentsFuture.thenComposeAsync(activeSegments ->
+                            notifyTxnCreation(scope, stream, activeSegments, txnId), executor).whenComplete((v, e) ->
+                            // Method notifyTxnCreation ensures that notification completes
+                            // even in the presence of n/w or segment store failures.
+                            log.debug("Txn={}, notified segments stores", txnId));
```

Comment:
Is this debug or trace level? If it is too specific and so that we can follow a particular code path, then better make it trace level.

Reply:
these are existing logs, and we have ample debug level logs in this flow and can make it trace